### ISSUE TITLE
Move hakyll-init into a separate package

### ISSUE THREAD

**Issue by sjakobi** [jaspervdj/hakyll#515]
Motivation:

* With very dependencies, hakyll-init would be very quick to install

* As [the installation instructions](https://jaspervdj.be/hakyll/tutorials/01-installation.html) currently stand, the `stack install hakyll` command to install hakyll-init doesn't necessarily use a similar Stackage snapshot as the stack project that contains the actual site. This means that many dependencies may have to be compiled twice leading to large disk space consumption.

* With very few dependencies, hakyll-init would be compatible with virtually any Stackage snapshot, making the command `stack install hakyll-init` unlikely to fail for compatibility reasons.

Alternatively, hakyll-init could be replaced or supplemented by a [stack template](https://github.com/commercialhaskell/stack-templates). The installation instructions could then look like this:

```
$ stack new my-site https://jaspervdj.be/hakyll/stack-template.hsfiles
$ cd my-site
$ stack build
```

**Comment by jaspervdj** [comment]
Yes, I think this is a good idea.  It slims down the package a bit.

**Comment by sjakobi** [comment]
Do you want to keep `hakyll-init` in this repo or should it be moved into a separate repo?

In the first case, hakyll-init's files should be moved into a new top-level directory `hakyll-init`, leaving all other files in their current place, right?

**Comment by jaspervdj** [comment]
I think it is a bit cleaner to do it in a separate repository.  I will set up a new empty repository and give you push access.

**Comment by sjakobi** [comment]
> I think it is a bit cleaner to do it in a separate repository.

The decision is up to you of course, but I think there's a downside to this approach:

Any feature or issue that affects both hakyll-the-library and hakyll-init will have to be tracked across two repos now, requiring i.e. two PRs instead of one. The increased troublesomeness of changing both packages might result in the two packages going out of sync and hakyll-init getting less love than needed.

OTOH I don't have much experience with library maintenance so maybe there's a strong upside to separating the repos too…

**Comment by jaspervdj** [comment]
Yeah, I realize that, and I've discussed the monorepository approach many times.  There's definitely huge wins, but since Hakyll is fairly mature, I don't expect that there will be many PRs that affect both libraries.

**Comment by sjakobi** [comment]
https://github.com/jaspervdj/hakyll-init is a complete package now (and has [a first bug](https://github.com/jaspervdj/hakyll-init/issues/1) too!)

TODOs:

* [ ] Enable Travis
* [ ] Not sure if the copyright year in the LICENSE needs to be updated
* [ ] Merge https://github.com/jaspervdj/hakyll/pull/516
* [ ] Once hakyll-init is released, update the installation instruction to recommend `stack install hakyll-init`

**Comment by jaspervdj** [comment]
I've decided to put this off for now.  It doesn't really have any advantages except for Stack users, where it offers slightly faster install times & less disk space.

However, for Stack users, I believe the templates solution is better anyway, so we should use that instead.  It seems like some changes are being made to the `hsfiles` format, so we'll see how that goes.

**Comment by sjakobi** [comment]
Ok! 

>  It seems like some changes are being made to the hsfiles format, so we'll see how that goes.

Do you have a link for that?

**Comment by jaspervdj** [comment]
The README for [stack-templates](https://github.com/commercialhaskell/stack-templates) says: 

> There are plans to change the way Stack templates are done in the near future. Therefore, the project is not accepting new templates at this moment.

**Comment by sjakobi** [comment]
This doesn't sound to me like they are planning to change the .hsfiles format. I think they simply want to move away from having a single central template repository at https://github.com/commercialhaskell/stack-templates and instead have package maintainers distribute their templates on Hackage (as discussed in https://github.com/commercialhaskell/stack/issues/2681).

In summary I think we can expect that the`stack new MY_PKG TEMPLATE_URL` syntax will still work with the same .hsfiles format in a few months.

Which means that we can move forward with creating a template for hakyll. I'll probably submit a PR to that effect in the next 2 to 3 weeks.

I'm not yet sure what to do about the haskell-logo.png file though. It doesn't seem as if stack templates (or mustache templates in general) support binary files. I have created [a StackOverflow question](http://stackoverflow.com/questions/42192180/how-can-i-embed-a-png-image-in-a-stack-mustache-template) to be sure. Maybe we can replace it with an SVG?

**Comment by sjakobi** [comment]
> It doesn't seem as if stack templates (or mustache templates in general) support binary files.

[They do!](http://stackoverflow.com/questions/42192180/how-can-i-embed-a-png-image-in-a-stack-mustache-template?noredirect=1#comment71551442_42192180)
